### DISCUSSION TITLE
Downgrading Keycloak fails

### DISCUSSION THREAD

**Discussion by souravs17031999** [keycloak/keycloak#14130] [Q&A] [answered]
Does Keycloak allow downgrade from a higher version to a lower version ?

**Comment by tkucuk** [comment]
Yes, I tried version downgrade from v19.0.1 to v17.0.0. I did not faced with any problem. I hope this is helpfull but you have to ensure that backup your db before starting upgrade.

**Reply by souravs17031999** [reply]
Did you started from v19.0.1 (with empty DB) and try to go back to v17.0.0 ?

It should fail due to change in store, I will share the logs here.

**Reply by Dhiraj-Jain** [reply]
Yes, we also tried downgrade from v22.0.5 to v22.0.1. We are facing the db issue. I'm thinking during upgrade, the keycloak db schema has been changed and when we try to connect to db with 22.0.1, it throws checksum error. Here is the error log
![MicrosoftTeams-image](https://github.com/keycloak/keycloak/assets/64320631/14aaf5e7-936b-4c2d-9e8f-a180eaa5a134)

**Comment by hmlnarik** [comment, marked as answer]
> Does Keycloak allow downgrade from a higher version to a lower version ?

No, not in the legacy store.

A N-1 to N version compatibility is planned for the new store.
